File: app/src/main/java/com/thathurleyguy/lightning/views/WemoButton.java
package com.thathurleyguy.lightning.views;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.Switch;
import android.widget.TextView;

import com.thathurleyguy.lightning.LightningService;
import com.thathurleyguy.lightning.R;
import com.thathurleyguy.lightning.models.WemoDevice;

import butterknife.ButterKnife;
import butterknife.InjectView;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public class WemoButton extends LinearLayout {
    @InjectView(R.id.switch_button) public Switch toggleSwitch;
    @InjectView(R.id.text_label) public TextView label;

    public WemoButton(Context context, final WemoDevice device) {
        super(context);
        inflate(context, R.layout.wemo_button, this);
        this.setLayoutParams(new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT, 1));
        ButterKnife.inject(this);

        label.setText(device.getName());
        toggleSwitch.setChecked(device.isPoweredOn());

        toggleSwitch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                LightningService.getService()
                        .toggleDevice(device.getId())
                        .subscribeOn(Schedulers.newThread())
                        .observeOn(AndroidSchedulers.mainThread())
                        .subscribe();
            }
        });
    }

    public void setChecked(boolean checked) {
        toggleSwitch.setChecked(checked);
    }
}

File: app/src/main/java/com/thathurleyguy/lightning/views/WemoButton.kt
package com.thathurleyguy.lightning.views

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.Switch
import android.widget.TextView

import com.thathurleyguy.lightning.LightningService
import com.thathurleyguy.lightning.R
import com.thathurleyguy.lightning.models.WemoDevice

import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

public class WemoButton(context: Context, device: WemoDevice) : LinearLayout(context) {
    init {
        View.inflate(context, R.layout.wemo_button, this)
        this.setLayoutParams(LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT, 1f))
        val toggleSwitch = findViewById(R.id.switch_button) as Switch
        val label = findViewById(R.id.text_label) as TextView

        label.setText(device.name)
        toggleSwitch.setChecked(device.poweredOn)

        toggleSwitch.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View) {
                LightningService.service.toggleDevice(device.id).subscribeOn(Schedulers.newThread()).observeOn(AndroidSchedulers.mainThread()).subscribe()
            }
        })
    }
}
